### PR TITLE
feat: introduced single sourcing for edge version

### PR DESCRIPTION
**Pull request by vickywane**
This closes this [issue](https://github.com/ambianic/ambianic-ui/issues/343)

This PR contains changes done to introduce single-sourcing on the edge version number used at runtime and also in API responses.

### CODE REVIEW

**Comment by vickywane** [comment]
@ivelin This is looking good for merging.

File: src/setup.py [hunk 1]
```diff
@@ -1,2 +1,5 @@
 from setuptools import setup
-setup()
+setup(
```
Comment (ivelin) [inline src/setup.py:2]: @vickywane we are using the [setup.cfg](https://github.com/ambianic/ambianic-edge/blob/master/src/setup.cfg) mechanism for declaring metadata including version. setup.py reads it automatically. No need to set name and version in setup.py. That breaks the way the semantic release script is tied to version updates.

**Review by ivelin — CHANGES_REQUESTED** [review]
Looks mostly good. Please add a few unit tests to cover the new features for core versioning and the updated status REST API.

**Comment by ivelin** [comment]
@vickywane not sure why LGTM triggers JavaScript analysis on this repo. Is it because of the change to the  .theia/settings.json  file? I saw the same issue in one of my new PRs and can't figure out how to turn it off. The LGTM docs point to menus that I can't find. Would be great if you can figure this out. Thank you!

**Comment by vickywane** [comment]
>@vickywane not sure why LGTM triggers JavaScript analysis on this repo. Is it because of the change to the .theia/settings.json file? I saw the same issue in one of my new PRs and can't figure out how to turn it off. The LGTM docs point to menus that I can't find. Would be great if you can figure this out. Thank you!

Removing `.theia` would be a fix.

**Comment by ivelin** [comment]
> > @vickywane not sure why LGTM triggers JavaScript analysis on this repo. Is it because of the change to the .theia/settings.json file? I saw the same issue in one of my new PRs and can't figure out how to turn it off. The LGTM docs point to menus that I can't find. Would be great if you can figure this out. Thank you!
> 
> Removing `.theia` would be a fix.

Well, that would mess up the gitpod environment settings.

**Comment by vickywane** [comment]
> > > @vickywane not sure why LGTM triggers JavaScript analysis on this repo. Is it because of the change to the .theia/settings.json file? I saw the same issue in one of my new PRs and can't figure out how to turn it off. The LGTM docs point to menus that I can't find. Would be great if you can figure this out. Thank you!
> > 
> > 
> > Removing `.theia` would be a fix.
> 
> Well, that would mess up the gitpod environment settings.

Okay, I didn't touch it. 

Can we ignore the test for now? 
I haven't found anything yet.

**Comment by vickywane** [comment]
@ivelin 

A fix for the Javascript LGTM would come in a separate PR. 
Seems I need to write a configuration file to configure it for python only.

**Comment by ivelin** [comment]
> > > > @vickywane not sure why LGTM triggers JavaScript analysis on this repo. Is it because of the change to the .theia/settings.json file? I saw the same issue in one of my new PRs and can't figure out how to turn it off. The LGTM docs point to menus that I can't find. Would be great if you can figure this out. Thank you!
> > > 
> > > 
> > > Removing `.theia` would be a fix.
> > 
> > 
> > Well, that would mess up the gitpod environment settings.
> 
> Okay, I didn't touch it.
> 
> Can we ignore the test for now?
> I haven't found anything yet.

Yes, we can ignore this particular check for now.

**Pull request merged**